### ISSUE TITLE
notifyd/notify_mailto.c doesn't use the smtpclient.c api

### ISSUE THREAD

**Issue by elliefm** [cyrusimap/cyrus-imapd#3065]
Maybe it should?

Right now it's hardcoded to fork and exec the IMAPOPT_SENDMAIL binary itself, which means that a) it ignores the IMAPOPT_SMTP_BACKEND option (which might be "host" or "sendmail", and b) even if the "sendmail" backend was chosen, it doesn't use the same implementation as the other things that send over SMTP

**Comment by brong** [comment]
Yeah, that seems reasonable. Giving this a "good first issue" tag, and leaving it open.
